packages/nextjs-mf/src/loaders/nextPageMapLoader.ts

```diff
@@ -6,8 +6,14 @@ import fs from 'fs';
 import { UrlNode } from '../../client/UrlNode';
 
 /**
- * Webpack loader which prepares MF map for NextJS pages
+ * Webpack loader which prepares MF map for NextJS pages.
+ * This function is the main entry point for the loader.
+ * It gets the options passed to the loader and prepares the pages map.
+ * If the 'v2' option is passed, it prepares the pages map using the 'preparePageMapV2' function.
+ * Otherwise, it uses the 'preparePageMap' function.
+ * Finally, it calls the loader's callback function with the prepared pages map.
  *
+ * @param {LoaderContext<Record<string, unknown>>} this - The loader context.
  */
 export default function nextPageMapLoader(
```

Comment:
The function 'exposeNextjsPages' can be improved by using a map function instead of a for loop. This can improve readability and maintainability.